libpoolprotocols/getwork/EthGetworkClient.cpp

```diff
@@ -53,13 +51,35 @@ void EthGetworkClient::disconnect()
 void EthGetworkClient::submitHashrate(string const & rate)
 {
 	// Store the rate in temp var. Will be handled in workLoop
+	// Hashrate submission does not need to be as quick as possible
 	m_currentHashrateToSubmit = rate;
+
 }
 
 void EthGetworkClient::submitSolution(Solution solution)
 {
-	// Store the solution in temp var. Will be handled in workLoop
-	m_solutionToSubmit = solution;
+	// Immediately send found solution without wait for loop
+	if (m_connected || m_justConnected) {
```

Comment:
What is the `m_justConnected` flag for?

Reply:
Really do not know. I only replicated what i believe is the "connected" condition imposed in workloop. I did not write this class.